### PR TITLE
feat: adding transactions for bulk post

### PR DESCRIPTION
**Pull request by manthanappveen**

### CODE REVIEW

File: crud.controller.js [hunk 1]
```diff
@@ -573,6 +587,8 @@ CrudController.prototype = {
         var upsert = reqParams['upsert'];
         var docIds = [];
         var body = params.map(req)[payload];
+        var restrictOnError = reqParams['restrictOnError'] && Array.isArray(body);
```
Comment (jerrymannel) [inline crud.controller.js]: The param is not `restrictOnError`. It is `batchMode`

File: crud.controller.js [hunk 2]
```diff
@@ -585,7 +601,17 @@ CrudController.prototype = {
             promise = self.model.find({ "_id": { "$in": docIds } });
         }
         return promise
-            .then(documents => {
+            .then(async documents => {
+                if (restrictOnError) {
+                    var startSession = await isTransactionSupported(self, res);
+                    if (startSession) {
+                        req.session = session = await self.model.startSession();
```
Comment (jerrymannel) [inline crud.controller.js:614]: We should define the [TransactionOptionsObject](https://mongodb.github.io/node-mongodb-native/3.6/api/global.html#TransactionOptions) for a session. This session that we are operating should define the _readPreference_, _writeConcern_ and _readConcern_.
It should be 
```
readPreference: Primary
write and readConcern: majority
```

**Review by jerrymannel — CHANGES_REQUESTED** [review]
@manthanappveen please check the review comments.

File: crud.controller.js (hunk 1 above)
Comment (jugnu-appveen) [reply crud.controller.js]: Hey, you told to make it `abortOnError` right?

**Pull request merged**